devops/actions/run-tests/benchmark/action.yml

```diff
@@ -37,59 +30,141 @@ inputs:
   exit_on_failure:
     type: string
     required: False
+  # Path to SYCL installation directory
+  sycl_dir:
+    type: string
+    required: False
+    default: "./toolchain"
+  # Only specific runners are supported
+  runner:
+    type: string
+    required: True
 
 runs:
-  # composite actions don't make use of 'name', so copy-paste names as a comment in the first line of each step
+  # composite actions don't make use of 'name', so copy-paste steps' names as a comment in the first line of each step
   using: "composite"
   steps:
-  - name: Check specified runner type / target backend
+  - name: Check inputs and set up environment
     shell: bash
     env:
+      # inputs are not directly used, as this allows code injection
       TARGET_DEVICE: ${{ inputs.target_devices }}
       PRESET: ${{ inputs.preset }}
+      SYCL_DIR: ${{ inputs.sycl_dir }}
+      RUNNER_TAG: ${{ inputs.runner }}
+      # Will append "_<device>_<backend>" to that prefix and use it as the full save name
+      SAVE_PREFIX: ${{ inputs.save_name }}
     run: |
+      # Check inputs and set up environment
+
+      # Ensure runner name has nothing injected
+      if [ -z "$(printf '%s' "$RUNNER_NAME" | grep -oE '^[a-zA-Z0-9_-]+$')" ]; then
+          echo "Bad runner name, please ensure runner name is [a-zA-Z0-9_-]."
+          exit 1
+      fi
+
       # Check specified runner type / target backend
       case "$RUNNER_TAG" in
-        '["PVC_PERF"]' ) ;;
-        '["BMG_PERF"]' ) ;;
+        '["PVC_PERF"]') GPU_TYPE="PVC" ;;
+        '["BMG_PERF"]') GPU_TYPE="BMG" ;;
         *)
+          # Best effort at matching if not known runners
+          # TODO: should we drop it and just exit instead?
+          GPU_TYPE="${RUNNER_TAG#[\"}"
+          GPU_TYPE="${GPU_TYPE%_PERF=\"]}"
           echo "#"
           echo "# WARNING: Only specific tuned runners are fully supported."
           echo "# This workflow is not guaranteed to work with other runners."
           echo "#" ;;
       esac
 
-      # Ensure runner name has nothing injected
-      # TODO: in terms of security, is this overkill?
-      if [ -z "$(printf '%s' "$RUNNER_NAME" | grep -oE '^[a-zA-Z0-9_-]+$')" ]; then
-          echo "Bad runner name, please ensure runner name is [a-zA-Z0-9_-]."
-          exit 1
-      fi
-
-      # input.target_devices is not directly used, as this allows code injection
       case "$TARGET_DEVICE" in
-        level_zero:*) ;;
-        level_zero_v2:*) ;;
+        level_zero:*)
+          SAVE_SUFFIX="L0"
+          ONEAPI_DEVICE_SELECTOR="level_zero:gpu"
+          ;;
+        level_zero_v2:*)
+          SAVE_SUFFIX="L0v2"
+          ONEAPI_DEVICE_SELECTOR="level_zero:gpu"
+          export SYCL_UR_USE_LEVEL_ZERO_V2=1
```

Comment:
You might want to explicitly set `SYCL_UR_USE_LEVEL_ZERO_V2=0` for the `level_zero:*)` case.